tsconfig.json

```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
+    "declaration": true,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "jsx": "react",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noFallthroughCasesInSwitch": true,
+    "noImplicitThis": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "outDir": "./lib",
+    "rootDir": "./src",
+    "skipLibCheck": true,
+    "sourceMap": true,
+    "target": "es2015"
```

Comment:
Could using es6 cause problems with older browsers? Maybe we still want to use babel to generate es5 and es6 compliant code? Or we just run babel after transpiling with tsc to get es5 compliant code (I do this in https://github.com/vega/vega-embed/blob/f9450f402da127fa3369ad12606b0964298c3a65/package.json#L69).

Reply:
I went with the original system to keep to just one output directory (`lib/`) and no intermediaries.

I also noticed that `Object.assign` doesn't get polyfilled in the output `lib/editor.js` on `master`, which is breaking this package on IE 11 😱 ... #293

Reply:
Actually, @domoritz, once #294's fix to #293 lands, this PR can have the target be `es5`, which would let the output run on browsers as old as IE 11. That'd be a bit cleaner IMO than continuing to use Babel. Thoughts?

Reply:
Sounds good. I merged #294